libs/common/Generator.cs

```diff
@@ -16,11 +16,22 @@ public static class Generator
         /// </summary>
         /// <param name="size">The length of the hex identifier string</param>
         /// <returns></returns>
-        public static string CreateHexId(int size = 40)
+        public static unsafe string CreateHexId(int size = 40)
         {
             Span<byte> nodeIdBuffer = stackalloc byte[size / 2];
             RandomNumberGenerator.Fill(nodeIdBuffer);
```

Comment:
Maybe add at least a debug assert or another arg validation if size is above 256 or so?

Reply:
yes, this would be great to add in this PR, thanks.

Reply:
is on public API, so can't make too many assumptions; now has suitable strategies to handle any reasonable input